system/lib/wasmfs/file_table.cpp

```diff
@@ -109,4 +147,19 @@ FileTable::Handle::Entry::operator bool() const {
 
   return fileTableHandle.fileTable.entries[fd] != nullptr;
 }
+
+__wasi_fd_t FileTable::Handle::add(std::shared_ptr<OpenFileState> openFileState) {
+  for (__wasi_fd_t i = 0; i < fileTable.entries.size(); i++) {
+    if (!fileTable.entries[i]) {
+      // Free open file entry.
+      fileTable.entries[i] = openFileState;
+      return i;
+    }
+  }
```

Comment:
Maybe add a todo to add a [free list](https://en.wikipedia.org/wiki/Free_list) for this, so we don't have linear lookup time. But that's not urgent.